### PR TITLE
[CIS-2295] Add support for hiding connection status with `isInvisible`

### PR DESCRIPTION
**Pull request by sgoodwin**
### 🔗 Issue Links

https://github.com/GetStream/stream-chat-swift/issues/2368
CIS-2295

### 🎯 Goal

The goal of this task is to be able to hide the connection status when connecting a user and also expose this property to the current user.

### 🛠 Implementation
- Adds `UserInfo.isInvisible` to be able to set the value when connecting the user
- Adds `CurrentUserDTO.isInvisible` to be able to get the value when the user is already connected

### 🧪 Manual Testing Notes
- Open a channel with User A
- With another device, before connecting User B, tap the Configuration Button in Demo App
- Enable `isInvisible` in the user info
- Open a channel with the User B
- User A should no see User B has online in the Channel Header View

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

**Comment by nuno-vieira** [comment]
Hi @sgoodwin!

Thank you a lot for your contribution 🙂 We will handle it from here 👍 I'm going to add a couple more tests and add the option to make yourself invisible when connecting.

Best,
Nuno

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

File: Sources/StreamChat/Models/CurrentUser.swift [hunk 1]
```diff
@@ -52,12 +52,16 @@ public class CurrentChatUser: ChatUser {
     
     /// The unread counts for the current user.
     public let unreadCount: UnreadCount
+
+    /// A Boolean value indicating if the user has opted to hide their online status.
+    public let isInvisible: Bool
```
Comment (polqf) [inline Sources/StreamChat/Models/CurrentUser.swift:57]: Are we sure this value will always exist? Would be nice to check with BE

Comment (nuno-vieira) [reply Sources/StreamChat/Models/CurrentUser.swift:57]: Yes, either way it has a default value of false

**Comment by sgoodwin** [comment]
Thank you all for this :)

**Pull request merged**